### ISSUE TITLE
Duplicate identifier wrongly reported between class declaration and expando function property assignment

### ISSUE THREAD

**Issue by dragomirtitian** [microsoft/typescript-go#1783]
## Steps to reproduce

```ts
declare class Logger {
    static instance(): void;
}

function Logger() { }

Logger.instance = function () { };
```
[Playground Link](https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEkoGdnwDIHsDm2Q3gG8BYAKHgvmQBcpqBLMeegOxqhbBAAoBKALngA3TPWABuMgF8yZAGYBXTg0wsMOPDD5F4M0mSy58AOlbtOCALzxFy+qvjbCu8UA)

## Behavior with `typescript@5.8`

No error, class declaration and expando function merge without error

## Behavior with `tsgo`

`Duplicate identifier 'instance'.` in the class declaration of `instance`  and in the assignment of `instance`

**Comment by ahejlsberg** [comment]
Strada has some dubious logic that sometimes treat expando properties as method declarations instead of property declarations. In Corsa, expando properties are always treated as property declarations. If you change the declaration of the `Logger` class to

```ts
declare class Logger {
    static instance: () => void;
}
```

then Strada errors, whereas Corsa doesn't. I think the best workaround here is to change the `Logger.instance`  assignment to

```ts
(Logger).instance = function () { };
```

With this change the assignment is no longer considered an expando property declaration. This also has the desirable effect of contextually typing the function expression by the type of the static property declaration.
